Design_Patterns.md

```diff
@@ -0,0 +1,74 @@
+# Design Patterns for Beginners
+
+## Introduction
+Design patterns are solutions to common problems encountered in software design. They provide a 'systematic' yet creative way to design and solve problems. When you're building software, you often face similar challenges, and design patterns offer a way to address these challenges in an efficient manner.
+
+### Why Design Patterns?
+
+1. **Reusability:** Design patterns promote code reusability. Once you understand and implement a design pattern, you can apply it to different projects and scenarios. Note that it is not a piece of code for you to copy and paste, but a certain way of thinking and solving problems. 
+   - But it is not the same as _algorithm_, whose input and output satisfy a certain specifications. Design Patterns are more creative and applicable in software development, in the sense that it does not require a strict specification on its input. 
+
+2. **Maintainability:** Using design patterns results in cleaner and more maintainable code. They encapsulate specific behaviors, making it easier to modify or extend functionalities without affecting the entire codebase.
+
+3. **Scalability:** Design patterns contribute to scalable and flexible software architecture. They help in building systems that can adapt to changing requirements and scale gracefully.
+
+
+## Types of Design Patterns

```

Comment:
From these 3 types, it seems like you are mentioning "Objects" a lot, maybe you want to clarify the relationship between OOP and Design Patterns? 

I believe that good OOP principles are tied to utilizing design patterns, but design patterns are not completely tied to OOP, you can also have design patterns in other programming languages.

Reply:
I have mentioned their relationship as well.